File: examples/python/basic-bot.py
import discord
from discord import app_commands
import os
from dotenv import load_dotenv

load_dotenv()
TOKEN = os.getenv('TOKEN')
PREFIX = os.getenv('PREFIX', '!')


class Bot(discord.Client):
    def __init__(self):
        intents = discord.Intents.default()
        intents.message_content = True
        super().__init__(intents=intents)
        self.tree = app_commands.CommandTree(self)

    async def on_ready(self):
        await self.tree.sync()
        print(f'Logged in as {self.user}')

    async def on_message(self, message):
        if message.author.bot:
            return
        if not message.content.startswith(PREFIX):
            return

        args = message.content[len(PREFIX):].split()
        if not args:
            return

        cmd = args[0].lower()
        if cmd == 'ping':
            await message.reply('Pong!')


bot = Bot()


@bot.tree.command(name='ping', description='Replies with Pong!')
async def ping(interaction: discord.Interaction):
    await interaction.response.send_message('Pong!')


bot.run(TOKEN)

File: examples/csharp/basic-bot.cs
using Discord;
using Discord.WebSocket;
using Discord.Net;
using System;
using System.Threading.Tasks;
using System.Linq;
using Newtonsoft.Json;

class Program
{
  private static DiscordSocketClient _client;
  private static string _prefix = "!";

  public static async Task Main(string[] args)
  {
    DotNetEnv.Env.Load();
    var token = Environment.GetEnvironmentVariable("TOKEN");
    _prefix = Environment.GetEnvironmentVariable("PREFIX") ?? "!";

    var config = new DiscordSocketConfig
    {
      GatewayIntents = GatewayIntents.AllUnprivileged | GatewayIntents.MessageContent
    };

    _client = new DiscordSocketClient(config);

    _client.Ready += OnReady;
    _client.SlashCommandExecuted += OnSlashCommand;
    _client.MessageReceived += OnMessage;

    await _client.LoginAsync(TokenType.Bot, token);
    await _client.StartAsync();

    await Task.Delay(-1);
  }

  private static async Task OnReady()
  {
    Console.WriteLine($"Logged in as {_client.CurrentUser.Username}");

    var command = new SlashCommandBuilder()
      .WithName("ping")
      .WithDescription("Replies with Pong!");

    try
    {
      await _client.CreateGlobalApplicationCommandAsync(command.Build());
      Console.WriteLine("Slash command registered");
    }
    catch (HttpException ex)
    {
      Console.WriteLine($"Failed to register command: {ex.Message}");
    }
  }

  private static async Task OnSlashCommand(SocketSlashCommand command)
  {
    if (command.Data.Name == "ping")
    {
      await command.RespondAsync("Pong!");
    }
  }

  private static async Task OnMessage(SocketMessage message)
  {
    if (message.Author.IsBot) return;
    if (!message.Content.StartsWith(_prefix)) return;

    var args = message.Content[_prefix.Length..].Split(' ', StringSplitOptions.RemoveEmptyEntries);
    if (args.Length == 0) return;

    switch (args[0].ToLower())
    {
      case "ping":
        await message.Channel.SendMessageAsync("Pong!");
        break;
    }
  }
}
